float32 x
float32 y
float32 z
float32 thetax
float32 thetay
float32 thetaz
---
bool output
